solarwinds_apm/oboe/settings.py

```diff
@@ -0,0 +1,223 @@
+
+    @sample_rate.setter
+    def sample_rate(self, new_sample_rate):
+        self._sample_rate = new_sample_rate
+
+    @property
+    def sample_source(self):
+        return self._sample_source
+
+    @sample_source.setter
+    def sample_source(self, new_sample_source):
+        self._sample_source = new_sample_source
+
+    @property
+    def flags(self):
+        return self._flags
+
+    @flags.setter
+    def flags(self, new_flags):
+        self._flags = new_flags
+
+    @property
+    def buckets(self):
+        return self._buckets
+
+    @buckets.setter
+    def buckets(self, new_buckets):
+        self._buckets = new_buckets
+
+    @property
+    def signature_key(self):
+        return self._signature_key
+
+    @signature_key.setter
+    def signature_key(self, new_signature_key):
+        self._signature_key = new_signature_key
+
+    @property
+    def timestamp(self):
+        return self._timestamp
+
+    @timestamp.setter
+    def timestamp(self, new_timestamp):
+        self._timestamp = new_timestamp
+
+    @property
+    def ttl(self):
+        return self._ttl
+
+    @ttl.setter
+    def ttl(self, new_ttl):
+        self._ttl = new_ttl
+
+    def __eq__(self, other):
+        if not isinstance(other, Settings):
+            return NotImplemented
+        return (
+            self._sample_rate == other._sample_rate
+            and self._sample_source == other._sample_source
+            and self._flags == other._flags
+            and self._buckets == other._buckets
+            and self._signature_key == other._signature_key
+            and self._timestamp == other._timestamp
+            and self._ttl == other._ttl
+        )
+
+    def __str__(self):
+        return f"Settings(sample_rate={self._sample_rate}, sample_source={self._sample_source}, flags={self._flags}, buckets={self._buckets}, timestamp={self._timestamp}, ttl={self._ttl})"
+
+
+class LocalSettings:
+    def __init__(self, tracing_mode: TracingMode | None, trigger_mode: bool):
+        self._tracing_mode = tracing_mode
+        self._trigger_mode = trigger_mode
+
+    @property
+    def tracing_mode(self):
+        return self._tracing_mode
+
+    @tracing_mode.setter
+    def tracing_mode(self, new_tracing_mode):
+        self._tracing_mode = new_tracing_mode
+
+    @property
+    def trigger_mode(self):
+        return self._trigger_mode
+
+    @trigger_mode.setter
+    def trigger_mode(self, new_trigger_mode):
+        self._trigger_mode = new_trigger_mode
+
+    def __eq__(self, other):
+        if not isinstance(other, LocalSettings):
+            return NotImplemented
+        return (
+            self._tracing_mode == other._tracing_mode
+            and self._trigger_mode == other._trigger_mode
+        )
+
+    def __str__(self):
+        return f"LocalSettings(tracing_mode={self._tracing_mode}, trigger_mode={self._trigger_mode})"
+
+
+def merge(
+    remote: Settings | None = None, local: LocalSettings | None = None
+) -> Settings | None:
```

Comment:
We don't need to add docstrings to every function (i.e. the properties/getters/setters are obvious enough) but adding them to some functions could help. Is this accurate?:

```suggestion
def merge(
    remote: Settings | None = None, local: LocalSettings | None = None
) -> Settings | None:
    """
    Returns remote sampling settings if available, else returns None. If possible, sets Flags by order of precedence (remote > local) unless remote has set an override.
    """
```